### DISCUSSION TITLE
Multiple different Android images?

### DISCUSSION THREAD

**Discussion by tgm-git** [waydroid/waydroid#1092] [Q&A] [answered]
I am very pleased with Waydroid so far. I have only encountered a small bug with the clipboard, but otherwise it has been great.

There is one feature I am currently missing from Waydroid. I have several apps I would like to run in a container for privacy reasons, such as messenger and facebook. There are also other android apps I want to use, but would like to them to be isolated from the Meta apps (such as my government id app).

Is it possible to run instances? My main knowledge of lxc and containerisation comes from docker, so in docker terms I want two or more containers running on the same base image. So I can have one container for Meta apps and another for personal apps.

From my understanding this should be possible with Waydroid, but I can't find anything about it in the docs. I have not had better luck and stack overflow, where the only posts I found where people using Waydroid for server containerisation.

Is this use-case possible with Waydroid?

**Comment by TriMoon** [comment]
IMHO, in the current state of Waydroid this is not possible.
Because they use fixed paths in their scripts and your homedir, which means you would need to modify a lot of stuff to get it to work as an extra independent instance.

If they ever make the Waydroid setup instance based, which uses directories OUTSIDE the users homedir it would make this possible, but right now it's just a single user setup...

FYI: They currently only use the base images from a directory outside the user's home, but the working version is fixed to a specific directory inside the user's home..

**Comment by ChrysoliteAzalea** [comment]
As far as I know, it's possible as long as only one instance is run at a time -- just put the ```/data``` directory you need to the ```~/.local/share/waydroid/data``` before running the container.

**Reply by tgm-git** [reply]
So I could create several separate data directories and then symlink to `~/.local/share/waydroid/data`?

This would allow me to separate apps from each other, which partially solves my problem, but the base image would still be the same I assume?

**Reply by ChrysoliteAzalea** [reply, marked as answer]
Yes, the base image is stored in ```/var/lib/waydroid/images```. You can still use separate Magisk modules in various instances of ```/data``` though.
